components/textbox/textbox.tsx

```diff
@@ -132,58 +143,66 @@ export default class Textbox extends React.PureComponent {
         }
     }
 
-    handleKeyDown = (e) => {
+    handleKeyDown = (e: KeyboardEvent) => {
         if (this.props.onKeyDown) {
             this.props.onKeyDown(e);
         }
     }
 
-    handleMouseUp = (e) => {
+    handleMouseUp = (e: MouseEvent) => {
         if (this.props.onMouseUp) {
             this.props.onMouseUp(e);
         }
     }
 
-    handleKeyUp = (e) => {
+    handleKeyUp = (e: KeyboardEvent) => {
         if (this.props.onKeyUp) {
             this.props.onKeyUp(e);
         }
     }
 
-    handleBlur = (e) => {
+    handleBlur = (e: FocusEvent) => {
         if (this.props.onBlur) {
             this.props.onBlur(e);
         }
     }
 
-    handleHeightChange = (height, maxHeight) => {
+    handleHeightChange = (height: number, maxHeight: number) => {
         if (this.props.onHeightChange) {
             this.props.onHeightChange(height, maxHeight);
         }
     }
 
     getInputBox = () => {
-        const textbox = this.refs.message.getTextbox();
-        return textbox;
+        if (this.message.current) {
+            return this.message.current.getTextbox();
```

Comment:
You can avoid the null check here by using the optional chaining operator.
```suggestion
            return this.message.current?.getTextbox();
```

Reply:
oh that's some pretty nice syntax